tests/pytorch/nightly/hf-llm.libsonnet

```diff
@@ -0,0 +1,147 @@
+    export PT_XLA_DEBUG=0
+    export USE_TORCH=ON
+  |||,
+
+  local command_copy_metrics = |||
+    gsutil -m cp -r /tmp/test-clm/*.json $(MODEL_DIR)
+  |||,
+
+  local gpt2_model = self.gpt2_model,
+  gpt2_model:: common.PyTorchTest {
+    local config = self,
+    modelName: 'hf-gpt2',
+    paramsOverride:: {
+      scriptPath: 'examples/pytorch/xla_spawn.py',
+      trainCommand: [
+        'python3',
+        self.scriptPath,
+        '--num_cores=4',
+        'examples/pytorch/language-modeling/run_clm.py',
+        '--num_train_epochs=3',
+        '--dataset_name=wikitext',
+        '--dataset_config_name=wikitext-2-raw-v1',
+        '--per_device_train_batch_size=%d ' % config.paramsOverride.per_device_train_batch_size,
+        '--per_device_eval_batch_size=%d ' % config.paramsOverride.per_device_eval_batch_size,
+        '--do_train',
+        '--do_eval',
+        '--logging_dir=./tensorboard-metrics',
+        '--cache_dir=./cache_dir',
+        '--output_dir=/tmp/test-clm',
+        '--overwrite_output_dir',
+        '--cache_dir=/tmp',
+        '--config_name=%s' % config.paramsOverride.config_name,
+        '--tokenizer_name=gpt2',
+        '--block_size=1024',
+        '--optim=adafactor',
+        '--adafactor=true',
+        '--save_strategy=no',
+        '--logging_strategy=no',
+        '--fsdp=full_shard',
+        '--fsdp_config=examples/pytorch/language-modeling/fsdp_config.json',
+      ],
+    },
+    command: utils.scriptCommand(
+      |||
+        %s
+        %s
+        %s
+      ||| % [
+        command_common,
+        utils.toCommandString(self.paramsOverride.trainCommand),
+        command_copy_metrics,
+      ]
+    ),
+  },
+
+
+  local config_2B = self.config_2B,
+  config_2B:: common.Convergence {
+    modelName: 'hf-gpt2-2b',
+    paramsOverride+:: {
+      config_name: '/home/xl-ml-test/transformers/examples/pytorch/language-modeling/my_config_2.json',
+      per_device_train_batch_size: 8,
+      per_device_eval_batch_size: 8,
+    },
+    metricConfig+: {
+      sourceMap+:: {
+        tensorboard+: {
+          aggregateAssertionsMap+:: {
+            'eval/loss': {
+              FINAL: {
+                fixed_value: {
+                  comparison: 'LESS',
+                  value: 7,
+                },
+                inclusive_bounds: false,
+                wait_for_n_data_points: 0,
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+
+  local tpuVm = self.tpuVm,
+  tpuVm:: common.PyTorchTpuVmMixin {
+    tpuSettings+: {
+      tpuVmExports+: |||
+        cd transformers/
+        export LD_LIBRARY_PATH=/usr/local/lib/
+        export PJRT_DEVICE=TPU_C_API
+        export PT_XLA_DEBUG=0
+        export USE_TORCH=ON
+      |||,
+      tpuVmExtraSetup: |||
+        pip install --upgrade accelerate
+        git clone https://github.com/huggingface/transformers.git
+        cd transformers
+        git checkout ebdb185befaa821304d461ed6aa20a17e4dc3aa2
```

Comment:
May I know the reason for that specific commit?

Reply:
the latest commit will have crash, so this is the stable version we can use.